### PR TITLE
TS-5034: Add a --enable-tsan configure option

### PR DESCRIPTION
**Pull request by bryancall**

### CODE REVIEW

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/1163/ for details.

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/1056/ for details.

File: configure.ac [hunk 1]
```diff
@@ -197,6 +197,15 @@ AC_ARG_ENABLE([asan],
 )
 AC_MSG_RESULT([$enable_asan])
 
+# Enable TSAN for the builds
+AC_MSG_CHECKING([whether to enable tsan])
+AC_ARG_ENABLE([tsan],
+  [AS_HELP_STRING([--enable-tsan],[turn on tsan])],
```
Comment (jpeach) [inline configure.ac]: "enable Thread Sanitizer"

Comment (bryancall) [inline configure.ac]: Sounds good - I will do an updated for this.

**Pull request merged**